core/src/main/java/io/neow3j/protocol/core/methods/response/NameState.java

```diff
@@ -4,20 +4,20 @@
 
 import java.util.Objects;
 
-public class NameState extends TokenState {
+public class NameState extends NFTokenState {
 
     @JsonProperty(value = "expiration")
-    private Integer expiration;
+    private long expiration;
```

Comment:
I think it's better to have complex types (in this case, `Long`), since the object has a `@JsonProperty`, and the JSON attribute "expiration" can be `null` for some weird reason. The goal is to not break the whole code because of a problematic serialization/deserialization. Keep this in mind for the future.

Reply:
You're right, I'll adapt that.